src/mca/gds/ds12/gds_dstore.c

```diff
@@ -2396,23 +2392,20 @@ inline pmix_status_t _dstore_fetch(const char *nspace, pmix_rank_t rank, const c
         kval_cnt = rinfo->count;
 
         /*  Initialize array for all keys of rank */
-        if ((NULL == key) || (kval_cnt > 0)) {
+        if ((NULL == key) && (kval_cnt > 0)) {
             kval = (pmix_value_t*)malloc(sizeof(pmix_value_t));
             if (NULL == kval) {
                 return PMIX_ERR_NOMEM;
             }
-
-            ninfo = kval_cnt;
-            PMIX_INFO_CREATE(info, ninfo);
-            if (NULL == info) {
-                rc = PMIX_ERR_NOMEM;
-                goto done;
-            }
-
-            PMIX_VALUE_CONSTRUCT(kval);
             kval->type = PMIX_DATA_ARRAY;
             kval->data.darray = (pmix_data_array_t*)malloc(sizeof(pmix_data_array_t));
             if (NULL == kval->data.darray) {
+                PMIX_VALUE_RELEASE(kval);
```

Comment:
I don't think you can `PMIX_VALUE_RELEASE(kval)` if you didn't `PMIX_VALUE_CONSTRUCT(kval)`